src/features/metrics/components/BallotSidebar.tsx

```diff
@@ -0,0 +1,37 @@
+"use client";
+
+import { useMemo } from "react";
+
+import { MetricWithProjects } from "~/utils/fetchMetrics";
+import { SidebarWithChart } from "./SidebarWithChart";
+import { api } from "~/utils/api";
+
+const parseBallotData = (ballot: MetricWithProjects[]) => {
+  const totalAmounts = ballot.reduce((sum, item) => sum + item.total, 0);
+
+  return ballot.map(({ id, name, total }) => ({
+    id,
+    name,
+    amount: total,
+    fraction: total / totalAmounts,
```

Comment:
Let's see if we can move this calculation to the backend in the `routers/metrics` file.